features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Apis/Details/BasicInfo.js

```diff
@@ -206,6 +215,16 @@ class BasicInfo extends Component {
         let promised_subscribe = api.subscribe(apiId, applicationId, policy);
         promised_subscribe.then(response => {
             console.log("Subscription created successfully with ID : " + response.body.subscriptionId);
+            this.addNotifications();
+            let applications = [];
```

Comment:
I think we could use `Array.filter` method to further simplify this.
Using [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method

```javascript
let applications = this.state.options.filter(application => subscribeApplicationId !== application.value );
```